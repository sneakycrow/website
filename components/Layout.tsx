import React, { Fragment, ReactElement } from 'react';
import Head from 'next/head';

type LayoutProps = {
  children: ReactElement<any> | ReactElement<any>[] | null
  title: string;
  description?: string;
}

const Layout = (props: LayoutProps) => (
  <Fragment>
    <Head>
      <title>{props.title}</title>
      <meta name="viewport" content="width=device-width, initial-scale=1" />
      <meta charSet="utf-8" />
      {props.description && (
        <meta name="description" content={props.description} />
      )}
    </Head>
    <main className="container mx-auto p-4">
      {props.children}
    </main>
  </Fragment>
);

export default Layout;
