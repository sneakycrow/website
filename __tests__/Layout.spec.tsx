import React from 'react';
import { render } from '@testing-library/react';
import Layout from '../components/Layout';

describe('<Component /> spec', () => {
  it('renders the component', () => {
    const renderResult = render(
      <Layout>
        <h1>Hello World</h1>
      </Layout>
    );

    expect(renderResult.container.firstChild).toMatchSnapshot();
  });
});
