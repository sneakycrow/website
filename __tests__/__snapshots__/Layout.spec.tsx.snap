// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Component /> spec renders the component 1`] = `
<div
  class="jsx-3539679555"
>
  <h1>
    Hello World
  </h1>
</div>
`;
